python/postprocessing/wmass/postproc.py

```diff
@@ -157,17 +185,26 @@ class bcolors:
                    WproducerModule()
                ]
     elif trigOnly: 
-        modules = [puWeightProducer(),preSelection(isMC=True, passall=passall, dataYear=dataYear, trigOnly=True)]
+        modules = [puWeightProducer_allData(),
+                   puWeightProducer(),
+                   preSelection(isMC=True, passall=passall, dataYear=dataYear, trigOnly=True)]
     else:
         modules = []
 else:
     if inputFile == '' : #this will run on the hardcoded file above     
         input_files.append( input_dir + ifileDATA )
     else : input_files.append( inputFile )
-    modules = [preSelection(isMC=isMC, passall=passall, dataYear=dataYear), 
-               ]
-    if jmeCorrections!=None: modules.insert(1,jmeCorrections())
-    if muonScaleRes!=None:   modules.insert(1, muonScaleRes())
+    if isTest:
```

Comment:
Another suggestion - Instead of adding a new sequence of modules, why don't you append to the existing sequence with the same if condition?

Reply:
Indeed at some point they should be added to the main sequence, once their usage is agreed. For now the purpose was only to test those specific modules in a test-like scenario, and I didn't want to disrupt the other sequences.

For another round, it could be beneficial to define a list of modules in the beginning of the script (possibly for each particular sequence), using the name in "name = lambda : moduleName(...)" defined in the relevant imported .py files, rather than using the constructor with all input arguments. And possibly some input options to add modules on the fly (at least for quick tests)